### PR TITLE
TextTruncate can handle null and undefined text inputs

### PR DESCRIPTION
**Pull request by TheRealBenjiLi**
**Jira:**

[PRTL-371](https://clever.atlassian.net/browse/PRTL-371)

**Overview:**
Flare 1070 was caused by a null/undefined being passed into TextTruncate. While it would make sense to avoid that happening in the first place, this change will prevent it at the component level too!

**Screenshots/GIFs:**
This is when null is passed in: 
Screen Shot 2021-04-07 at 4 45 33 PM

Compared to normal: 
Screen Shot 2021-04-07 at 4 45 55 PM

**Testing:**

- [ ] Unit tests
- Manual tests:
  - [x] Chrome
  - [ ] Safari
  - [ ] IE11

**Roll Out:**

- Before merging:
  - [ ] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add an entry in `ComponentsView.componentsToDisplay` using this template:
      ```
      {
        componentLink: "<COMPONENT LINK>",
        componentImg: "<COMPONENT LINK>.png",
        componentName: "<COMPONENT NAME>",
        componentImgAlt: "A <COMPONENT NAME> component",
      },
      ```
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/TextTruncate/TextTruncate.tsx [hunk 1]
```diff
@@ -72,6 +72,13 @@ export default class TextTruncate extends React.PureComponent<Props> {
     } = this.props;
     const { truncated } = this.state;
 
+    // If text is null or undefined, return immediately
+    if (!text) {
+      return (
+        <></>
```
Comment (prime-time) [inline src/TextTruncate/TextTruncate.tsx:78]: I would actually return `null` here since fragments are typically used to wrap other elements.

**Review by prime-time — APPROVED** [review]
🔧

File: src/TextTruncate/TextTruncate.tsx (hunk 1 above)
Comment (prime-time) [reply src/TextTruncate/TextTruncate.tsx:78]: I should add that this is actually the recommended way as opposed to my personal preference 😄 
https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

Comment (TheRealBenjiLi) [reply src/TextTruncate/TextTruncate.tsx:78]: For sure! I didn't know null was an acceptable return here, since it's for rendering. Thanks! :)

**Pull request merged**